core/domain/classifier_services.py

```diff
@@ -118,3 +122,68 @@ def classify_string_classifier_rule(state, normalized_answer):
             return None
 
     return None
+
+
+def get_classifier_from_model(classifier_model):
+    """Returns a classifier domain object given a classifier model loaded
+    from datastore.
+    """
+    return classifier_domain.Classifier(
+        classifier_model.id, classifier_model.exp_id,
+        classifier_model.exp_version_when_created,
+        classifier_model.state_name, classifier_model.algorithm_id,
+        classifier_model.cached_classifier_data,
+        classifier_model.data_schema_version)
+
+def get_classifier_by_id(classifier_id):
+    """Returns a classifier domain object given a classifier id.
+    """
+    classifier_model = classifier_models.ClassifierModel.get(
+        classifier_id, strict=False)
+    if classifier_model is None:
+        raise Exception("No classifier found for the classifer's id.")
+    else:
+        classifier = get_classifier_from_model(classifier_model)
+        return classifier
+
+
+def _create_classifier(classifier):
+    """Creates classifier model in the datastore given a classifier
+       domain object.
+    """
+    classifier_models.ClassifierModel.create(
+        classifier.exp_id, classifier.exp_version_when_created,
+        classifier.state_name, classifier.algorithm_id,
+        classifier.cached_classifier_data, classifier.data_schema_version,
+        )
+
+
+def _save_classifier(classifier_model, classifier):
+    """Updates classifier model in the datastore given a classifier
+    domain object.
+    """
+    classifier_model.state_name = classifier.state_name
+    classifier_model.put()
+
+
+def update_classifier(classifier):
+    """Checks if model exists and updates the classifier model using
+    _save_classifier method.
+    """
+    classifier_model = classifier_models.ClassifierModel.get(
+        classifier.id, strict=False)
+    if classifier_model is None:
+        raise Exception("No classifier found for the classifer's id.")
```

Comment:
@seanlip I'm not sure if I understood the use case totally. There can be two things that we might need - (i) the exploration (and the state within it) to which this classifier belongs (ii) the latest version of exploration to which classifier belongs. I don't think in the second case we should expect that the state name would not have changed. I don't even think that there will be any use of such a "lookup". Do you have any specific scenario in your mind?